proto/api.proto

```diff
@@ -660,10 +672,35 @@ message PostSubmitRequest {
   bool skipPreFlight = 2;
 }
 
+message PostSubmitBatchRequestEntry {
+  string transaction = 1;
+  bool skipPreFlight = 2;
+}
+
+enum SubmitStrategy {
+  P_SUBMIT_ALL = 0;
+  P_ABORT_ON_FIRST_ERROR = 1;
+  P_ALWAYS_EXECUTE_LAST = 3;
+}
+
+message PostSubmitBatchRequest {
+  repeated PostSubmitBatchRequestEntry entries = 1;
+  SubmitStrategy SubmitStrategy = 2;
+}
+
 message PostSubmitResponse {
   string signature = 1;
 }
 
+message PostSubmitBatchResponseEntry {
+  string signature = 1;
+  string error = 2;
```

Comment:
perhaps also a boolean that indicates if it's submitted or not? could encapsulate that in error as well –– wdyt?